managed/services/grafana/client.go

```diff
@@ -223,26 +227,28 @@ var emptyUser = authUser{
 // getAuthUser returns grafanaAdmin if currently authenticated user is a Grafana (super) admin.
 // Otherwise, it returns a role in the default organization (with ID 1).
 // Ctx is used only for cancelation.
-func (c *Client) getAuthUser(ctx context.Context, authHeaders http.Header) (authUser, error) {
+func (c *Client) getAuthUser(ctx context.Context, authHeaders http.Header, l *logrus.Entry) (authUser, error) {
 	// Check if API Key or Service Token is authorized.
 	token := auth.GetTokenFromHeaders(authHeaders)
 	if token != "" {
 		role, err := c.getRoleForServiceToken(ctx, token)
-		if err != nil {
-			if strings.Contains(err.Error(), "Auth method is not service account token") {
-				role, err := c.getRoleForAPIKey(ctx, authHeaders)
-				return authUser{
-					role:   role,
-					userID: 0,
-				}, err
-			}
+		if err == nil {
+			return authUser{
+				role:   role,
+				userID: 0,
+			}, nil
+		}
+		if errors.Is(err, ErrIsNotServiceAccount) {
+			l.Warning("you should migrate your API Key into Service Account")
```

Comment:
let's show this warning after checking if it's API key. otherwise user will see this error for incorrect service token too.

Reply:
Good point, thanks. Now warning will be visible only if it is API key.